core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -993,6 +1024,26 @@ private String printDynamicFilters(Collection<DynamicFilters.Descriptor> filters
                     .collect(Collectors.joining(", ", "{", "}"));
         }
 
+        private String printDynamicFilterDomainStats(Collection<DynamicFilters.Descriptor> filters)
+        {
+            return filters.stream()
+                    .map(DynamicFilters.Descriptor::getId)
+                    .distinct()
+                    .map(id -> {
+                        DynamicFilterDomainStats stats = dynamicFilterDomainStats.get(id);
+                        if (stats == null) {
+                            return "";
```

Comment:
This can ends up as 
```
Dynamic filters: {}
```

line in EXPLAIN ANALYZE output, basically for every node within a plan, even for which one wouldn't expect dynamic filters.

eg. consider 
```
- UNION
   - JOIN
      - TS1
      - TS2
  - AGG
    - TS3
```

the fact that there is some Join somewhere in the plan (a global property) should not impact how TS3 is represented in the EXPLAIN ANALYZE output, should it?

Reply:
I'm using this function as
```
if (!dynamicFilters.isEmpty() && !dynamicFilterDomainStats.isEmpty()) {
       nodeOutput.appendDetailsLine("Dynamic filters: %s", printDynamicFilterDomainStats(dynamicFilters));
}
```
This should ensure that the TS actually has a dynamic filter and some domain stats exist.
I can further tighten it up to check for non-empty intersection between `dynamicFilters` and `dynamicFilterDomainStats` to take care of all possibilities.

Reply:
i read `!dynamicFilters.isEmpty() && !dynamicFilterDomainStats.isEmpty()` as a check on a global property. did i read it incorrectly?

Reply:
I pulled out `dynamicFilters` from current FilterNode within `visitScanFilterAndProjectInfo`.
`dynamicFilterDomainStats` is a global map, but we pull out stats relevant for current FilterNode by looking up the relevant dynamic filter ids based on `dynamicFilters`.